### PR TITLE
compress copy 5

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
Using compress_copy instead of compress_store
I removed one version of an example since meh, there is another version.
Added a trait to algo.

### CODE REVIEW

File: include/eve/module/algo/algo/remove.hpp [hunk 1]
```diff
@@ -31,8 +31,9 @@ namespace eve::algo
       EVE_FORCEINLINE bool step(auto it, eve::relative_conditional_expr auto ignore, auto /*idx*/)
       {
         auto loaded = eve::load[ignore](it);
-        auto mask   = !p(loaded);  // we decied that ! can be optimized well enough
-        out = compress_store[unsafe][ignore](loaded, mask, out);
+        auto mask   = !p(loaded); // we decied that ! can be optimized well enough
+        auto density   = denisty_for_compress_copy<typename TraitsSupport::traits_type>();
```
Comment (jfalcou) [inline include/eve/module/algo/algo/remove.hpp:35]: `density_for_compress_copy` I guess ?

File: examples/algorithms/writing_new/collect_indexes__complicated_real_example.cpp [hunk 2]
```diff
@@ -81,26 +79,28 @@ void collect_indexes(R&& r, P p, std::vector<IdxType, Alloc>& res)
       r_with_idx,
       [&](eve::algo::iterator auto r_idx_it, eve::relative_conditional_expr auto ignore) mutable
       {
+        auto [elems_it, idxs_it] = r_idx_it;
```
Comment (jfalcou) [inline examples/algorithms/writing_new/collect_indexes__complicated_real_example.cpp:82]: why the change ?

File: include/eve/module/algo/algo/set_intersection.hpp [hunk 3]
```diff
@@ -257,7 +257,8 @@ namespace detail
         auto v2 = eve::load(f2);
 
         auto matches = eve::has_equal_in(v1, v2, equal);
-        o            = eve::compress_store[eve::unsafe](v1, matches, o);
+        auto density = denisty_for_compress_copy<decltype(modified_traits())>();
```
Comment (jfalcou) [inline include/eve/module/algo/algo/set_intersection.hpp:260]: `density_for_compress_copy`

File: test/unit/module/algo/traits.cpp [hunk 4]
```diff
@@ -173,6 +173,32 @@ TTS_CASE("eve.algo.traits, expect_smaller_range")
   }
 };
 
+TTS_CASE("eve.algo.traits, denisty_for_compress_copy")
+{
+  {
+    eve::algo::traits tr {eve::algo::dense_output};
+    const auto density = eve::algo::denisty_for_compress_copy<decltype(tr)>();
```
Comment (jfalcou) [inline test/unit/module/algo/traits.cpp:180]: `density_for_compress_copy`

File: examples/algorithms/writing_new/collect_indexes__complicated_real_example.cpp (hunk 2 above)
Comment (DenisYaroshevskiy) [reply examples/algorithms/writing_new/collect_indexes__complicated_real_example.cpp:82]: Because we now need the idx iterator. `eve::compress_copy` is better than `eve::compress_store` since it has more opportunities to be optimized

**Review by jfalcou — APPROVED** [review]
